tests/virtiofs/config.go

```diff
@@ -0,0 +1,272 @@
+
+	"kubevirt.io/kubevirt/tests/exec"
+	"kubevirt.io/kubevirt/tests/testsuite"
+
+	"kubevirt.io/kubevirt/pkg/config"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+var _ = Describe("[sig-compute] vitiofs config volumes", decorators.SigCompute, func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+		checks.SkipTestIfNoFeatureGate(virtconfig.VirtIOFSGate)
+	})
+
+	Context("With a single ConfigMap volume", func() {
+		var (
+			configMapName string
+			configMapPath string
+		)
+
+		BeforeEach(func() {
+			// We use the ConfigMap name as mount `tag` for qemu, but the `tag` property must be 36 bytes or less
+			configMapName = "configmap-" + uuid.NewRandom().String()[:6]
+			configMapPath = config.GetConfigMapSourcePath(configMapName)
+
+			data := map[string]string{
+				"option1": "value1",
+				"option2": "value2",
+				"option3": "value3",
+			}
+			tests.CreateConfigMap(configMapName, testsuite.NamespacePrivileged, data)
+		})
+
+		It("Should be the mounted virtiofs layout the same for a pod and vmi", func() {
+			expectedOutput := "value1value2value3"
+
+			By("Running VMI")
+			vmi := libvmi.NewFedora(
+				libvmi.WithConfigMapFs(configMapName, configMapName),
+				libvmi.WithNamespace(testsuite.NamespacePrivileged),
+			)
+			vmi = tests.RunVMIAndExpectLaunchIgnoreWarnings(vmi, 300)
+
+			By("Logging into the VMI")
+			Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+			By("Checking if ConfigMap has been attached to the pod")
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+			podOutput, err := exec.ExecuteCommandOnPod(
+				virtClient,
+				vmiPod,
+				fmt.Sprintf("virtiofs-%s", configMapName),
+				[]string{"cat",
+					configMapPath + "/option1",
+					configMapPath + "/option2",
+					configMapPath + "/option3",
+				},
+			)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(podOutput).To(Equal(expectedOutput))
+
+			By("Checking mounted ConfigMap")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: fmt.Sprintf("mount -t virtiofs %s /mnt \n", configMapName)},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "echo $?\n"},
+				&expect.BExp{R: console.RetValue("0")},
+				&expect.BSnd{S: "cat /mnt/option1 /mnt/option2 /mnt/option3\n"},
+				&expect.BExp{R: expectedOutput},
+			}, 200)).To(Succeed())
+		})
+	})
+
+	Context("With a single Secret volume", func() {
+		var (
+			secretName string
+			secretPath string
+		)
+
+		BeforeEach(func() {
+			// We use the Secret name as mount `tag` for qemu, but the `tag` property must be 36 bytes or less
+			secretName = "secret-" + uuid.NewRandom().String()[:6]
+			secretPath = config.GetSecretSourcePath(secretName)
+
+			data := map[string]string{
+				"user":     "admin",
+				"password": "redhat",
+			}
+			tests.CreateSecret(secretName, testsuite.NamespacePrivileged, data)
```

Comment:
Will this be cleaned?

Reply:
As far as I was able to test, both the secret and the namespace are cleaned up. Those are the steps followed for testing this:

1. Focused the `With a single Secret volume` functest.
2. Targeted `make functest`
3. Ran `kubectl get namespaces. The command does not output `kubevirt-test-privileged` namespace.
4. Ran `kubectl get secrets --all-namespaces`. The command does not output any `secret-[random]` secret.

Therefore, we are convinced that both the secret and namespace are cleaned up after the test is executed.

Reply:
Right they are namespaced but you could still get collision. Nevertheless, we can look at this later.